tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -91,6 +91,27 @@ public void AddSpan(Span span)
             }
         }
 
+        public void CloseServerlessSpan()
```

Comment:
I'm not sure I understand this one. I see you're not passing `span` like in `CloseSpan(Span)`. Is this the dummy root span from the comments? I think this assumes that the root span is the last to close, which is not always true: child span can be an async operation that finished _after_ the root span.

And what about sampling priority? This method is not adding it to the root span (or to all spans if partial flush is enabled). Is sampling priority not used in Serverless?

Reply:
Thanks for clarifying offline, @maxday! I'll try to explain here for posterity:

The Lambda extension already creates a root span for the request, outside of the .NET Tracer. You want all spans created by the .NET Tracer to be children of that root span. To achieve this, you create a "fake" root span in the .NET Tracer with the same traceId and spanId as the real root span. Later, when that fake root span finishes, you _don't_ add it to the outgoing queue so it's not sent to the Agent.

```
[===========================] <- real root span created by extension (outside of .NET Tracer)
 [=========================]  <- fake root created by .NET Tracer (do not send to Agent!)
    [===]     [======]        <- instrumentation by .NET Tracer
```

Someday we may have a flag that says "don't serialize or send this span to the Agent" or a span processor that we can use to skip spans. When that happens, we can use that to skip this fake root span in a cleaner fashion.